system-addon/test/unit/lib/SectionsManager.test.js

```diff
@@ -493,5 +515,52 @@ describe("SectionsFeed", () => {
       assert.calledOnce(stub);
       assert.calledWith(stub, "Foo", "bar.com");
     });
+    it("should call the feed's uninit on UNINIT", () => {
+      sinon.stub(feed, "moveSection");
+      const id = "topsites";
+      const direction = +1;
+      feed.onAction({type: "SECTION_MOVE", data: {id, direction}});
+
+      assert.calledOnce(feed.moveSection);
+      assert.calledWith(feed.moveSection, id, direction);
+    });
+  });
+  describe("#moveSection", () => {
```

Comment:
can you add a test to make sure items are moved correctly when there is a disabled item inbetween?

e.g. if your sections are:

```
 [{id: "A", enabled: true}, {id: "B", enabled: false}, {id: "C", enabled: true}]
```

and you move `C` up, the order should be `C`, `A`, `B`, not `A`, `C`, `B`